python/cugraph/structure/graph.py

```diff
@@ -471,13 +503,21 @@ def number_of_vertices(self):
 
         """
         if self.node_count is None:
-            if self.adjlist is not None:
+            if self.distributed:
+                if self.edgelist is not None:
```

Comment:
If an adj list is present I assume we could just get it from there too.

Reply:
Currently we don't allow initialization of distributed graphs using adjlist. So for a distributed graph, if it is not empty,  would mean an edgelist is always present.